### PR TITLE
#1350: 2.0.x backport of flexible update hook numbering scheme implementation.

### PR DESCRIPTION
**Pull request by joeparsons**
## Description
2.0.x backport of #1351.

- Adds update hook numbering guidelines to RELEASES.md
- Adds empty az_quickstart_update_9209() to 2.0.x branch for #1350.
- Moves the existing update hooks from `.profile` to `.install` (as is being done in #1351) to prevent performance impact on non update/install requests.

Note this is (intentionally) targeted at the `2.0.x` branch (instead of `main`).

## Related Issue
#1350.

## How Has This Been Tested?

Locally with lando.

## Types of changes

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:

- [x] I've added this Pull Request to the AZ Quickstart Github project and set it to "Review in progress".
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

### CODE REVIEW

**Review by trackleft — APPROVED** [review]
Tested this by installing main branch of az_quickstart, and then switching to the issue/1350-2.0.x branch. No updates available.

**Comment by joeparsons** [comment]
> Tested this by installing main branch of az_quickstart, and then switching to the [issue/1350](https://github.com/az-digital/az_quickstart/issues/1350)-2.0.x branch. No updates available.

Thanks for testing!  FWIW, I think a more useful test would be to install Quickstart after checking out an existing release tag (e.g. 2.0.18) and then checkout this branch to see what updates are available.  `9209` _should_ show up as an available update for existing 2.0.x sites, even though it's empty and doesn't actually do anything.

**Comment by trackleft** [comment]
Retested by checking out Quickstart 2.0.18
Install Quickstart
`git checkout issue/1350-2.0.x`
`lando ssh`
`composer update`
`lando drush updb`
image

File: az_quickstart.install [hunk 1]
```diff
@@ -49,3 +49,104 @@ function az_quickstart_install() {
...
+    return t('Pantheon Advanced Page Cache module installed.');
+  }
+}
+
+/**
+ * Quickstart 2.1.x only update (removed).
+ */
+function az_quickstart_update_9209() {
+  // Update removed and moved to az_quickstart_update_92101().
+  // See https://github.com/az-digital/az_quickstart/issues/1350.
```
[the remark this answers is not in the document]
Comment (joeparsons) [reply az_quickstart.install:151]:
```suggestion
  // @see https://github.com/az-digital/az_quickstart/issues/1350
```

File: az_quickstart.install [hunk 2]
```diff
@@ -49,3 +49,104 @@ function az_quickstart_install() {
...
+
+    return t('Pantheon Advanced Page Cache module installed.');
+  }
+}
+
+/**
+ * Quickstart 2.1.x only update (removed).
+ */
+function az_quickstart_update_9209() {
+  // Update removed and moved to az_quickstart_update_92101().
```
Comment (joeparsons) [inline az_quickstart.install:150]:
```suggestion
  // Update removed and moved to az_quickstart_update_920101().
```

**Comment by joeparsons** [comment]
## 2022-03-02 Updates

- Added update hook numbering convention guidelines to RELEASES.md

**This should be ready for final review/approval**.

Reminder:  This PR is (intentionally) targeted at the `2.0.x` branch (instead of `main`).

**Review by trackleft — APPROVED** [review]

**Review by tadean — APPROVED** [review]

**Pull request merged**